### PR TITLE
feat: Implements Position Timeline component in org

### PR DESCRIPTION
**Pull request by tobiasny**

### CODE REVIEW

File: src/components/org/PositionTimeline/SelectableTimeline/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            initialSplit={initialSplit}
+            initialDate={initialDate}
+            previewDates={previewDates}
+            onDateChange={onDateChange}
+            onSplitSelect={onSplitSelect}
+            highlighted={highlighted}
+            disabled={disabled}
+        >
+            <RotationSequence
+                PersonSlot={PersonSlot}
```
Comment (eskil0312) [inline src/components/org/PositionTimeline/SelectableTimeline/index.tsx:45]: Should the user of the component have the opportunity to provide these components them self?

File: src/components/org/PositionTimeline/SelectableTimeline/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                InfoSlot={InfoSlot}
```
Comment (eskil0312) [inline src/components/org/PositionTimeline/SelectableTimeline/index.tsx:46]: I see these components are passed as props, can they be added to the context instead?

File: src/components/org/PositionTimeline/SelectableTimeline/index.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    initialSplit?: TimelineSplit;
+    initialDate?: Date;
+    previewDates?: boolean;
+    onDateChange?: (date: Date) => void;
+    onSplitSelect?: (split: string) => void;
+    highlighted?: string[];
+    disabled?: string[];
+};
+
+export const SelectableTimeline: FC<SelectableTimelineProps> = ({
```
[the remark this answers is not in the document]
Comment (tobiasny) [reply src/components/org/PositionTimeline/SelectableTimeline/index.tsx]: Renamed PositionTimeline and moved up one level in the directory.

File: src/components/org/PositionTimeline/Split/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { clsx } from '@equinor/fusion-react-styles';
```
[the remark this answers is not in the document]
Comment (tobiasny) [reply src/components/org/PositionTimeline/Split/index.tsx]: Components moved to a separate /components folder.

File: src/components/org/PositionTimeline/SelectableTimeline/index.tsx (hunk 1 above)
Comment (tobiasny) [reply src/components/org/PositionTimeline/SelectableTimeline/index.tsx]: Custom slot components may now be provided as props through the wrapper component.

File: src/components/org/PositionTimeline/SelectableTimeline/index.tsx (hunk 2 above)
Comment (tobiasny) [reply src/components/org/PositionTimeline/SelectableTimeline/index.tsx]: Custom slot components are now handled by context provider, which mitigates the propagation of props.

File: src/components/org/PositionTimeline/model.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    item: T;
+};
+
+export type InfoSlotProps<T> = {
+    item: T;
+};
+
+export type ActionSlotProps<T> = {
+    value: T;
+};
```
Comment (eskil0312) [inline src/components/org/PositionTimeline/model.ts:57]: Could this be one type?

Comment (eskil0312) [reply src/components/org/PositionTimeline/model.ts:57]: Does this need to be generic? Could T be TimelineSplit in all cases?

File: src/index.ts [hunk 6]
```diff
@@ -163,6 +163,7 @@ export {
     OrgChartItemProps,
     OrgNode,
 } from './components/org/OrgChart';
+export { default as PositionTimeline } from './components/org/PositionTimeline';
```
Comment (eskil0312) [inline src/index.ts:166]: Should export the slot type as well

**Comment by eskil0312** [comment]
Did you manage to downgrade node to 14 and update package-json lock

File: src/components/org/PositionTimeline/model.ts (hunk 5 above)
Comment (tobiasny) [reply src/components/org/PositionTimeline/model.ts]: Replaced by a common prop type with a split and isSelected prop.

File: src/index.ts (hunk 6 above)
Comment (tobiasny) [reply src/index.ts:166]: Exporting slotprop type from index.

**Review by eskil0312 — APPROVED** [review]

**Pull request merged**